From ae2804fa3db4eaaa89e8d49483d4ea2a8199d399 Mon Sep 17 00:00:00 2001
From: Karthik Kambatla <kasha@cloudera.com>
Date: Thu, 9 May 2013 15:23:59 -0700
Subject: [PATCH 058/596] MR1: CLOUDERA-BUILD. CDH-9234. Fix API
 incompatibility in
 mapredue.lib.jobcontrol.ControlledJob (cherry
 picked from commit
 53024f9fd2a32e2fabc6f8b499bb6fda5c3c01c6)

(cherry picked from commit 65608fd9f474a30faf684baf5e60a62ada0815fd)
(cherry picked from commit 030e51c5ede62ce4e6b2f6a39f54f3725a39a281)
(cherry picked from commit 570eafd53557bd34282f576fdc3372125b232181)
---
 .../hadoop/mapred/jobcontrol/JobControl.java       |    3 ++-
 .../mapreduce/lib/jobcontrol/ControlledJob.java    |    9 +++++----
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/jobcontrol/JobControl.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/jobcontrol/JobControl.java
index 672ce3d..3c89235 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/jobcontrol/JobControl.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/jobcontrol/JobControl.java
@@ -19,6 +19,7 @@
 package org.apache.hadoop.mapred.jobcontrol;
 
 import java.util.ArrayList;
+import java.util.List;
 import java.util.Collection;
 
 import org.apache.hadoop.mapreduce.lib.jobcontrol.ControlledJob;
@@ -39,7 +40,7 @@ public JobControl(String groupName) {
     super(groupName);
   }
   
-  static ArrayList<Job> castToJobList(ArrayList<ControlledJob> cjobs) {
+  static ArrayList<Job> castToJobList(List<ControlledJob> cjobs) {
     ArrayList<Job> ret = new ArrayList<Job>();
     for (ControlledJob job : cjobs) {
       ret.add((Job)job);
diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/ControlledJob.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/ControlledJob.java
index ca2a2ad..a67dd83 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/ControlledJob.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapreduce/lib/jobcontrol/ControlledJob.java
@@ -21,6 +21,7 @@
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.List;
 
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.FileSystem;
@@ -54,14 +55,14 @@
   // some info for human consumption, e.g. the reason why the job failed
   private String message;
   // the jobs the current job depends on
-  private ArrayList<ControlledJob> dependingJobs;
+  private List<ControlledJob> dependingJobs;
 	
   /** 
    * Construct a job.
    * @param job a mapreduce job to be executed.
    * @param dependingJobs an array of jobs the current job depends on
    */
-  public ControlledJob(Job job, ArrayList<ControlledJob> dependingJobs) 
+  public ControlledJob(Job job, List<ControlledJob> dependingJobs)
       throws IOException {
     this.job = job;
     this.dependingJobs = dependingJobs;
@@ -188,7 +189,7 @@ public synchronized void setMessage(String message) {
   /**
    * @return the depending jobs of this job
    */
-  public ArrayList<ControlledJob> getDependentJobs() {
+  public List<ControlledJob> getDependentJobs() {
     return this.dependingJobs;
   }
   
@@ -227,7 +228,7 @@ public synchronized boolean isReady() {
     return this.state == State.READY;
   }
 
-  public void killJob() throws IOException {
+  public void killJob() throws IOException, InterruptedException {
     job.killJob();
   }
   
-- 
1.7.9.5

